cmd/executor/cmd/root.go

```diff
@@ -110,21 +113,40 @@ func checkDockerfilePath() error {
 	return errors.New("please provide a valid path to a Dockerfile within the build context")
 }
 
-// resolveSourceContext unpacks the source context if it is a tar in a GCS bucket
+// resolveSourceContext unpacks the source context if it is a tar in a bucket
 // it resets srcContext to be the path to the unpacked build context within the image
 func resolveSourceContext() error {
 	if srcContext == "" && bucket == "" {
-		return errors.New("please specify a path to the build context with the --context flag or a GCS bucket with the --bucket flag")
+		return errors.New("please specify a path to the build context with the --context flag or a bucket with the --bucket flag")
 	}
-	if bucket == "" {
-		return nil
-	}
-	logrus.Infof("Using GCS bucket %s as source context", bucket)
+
 	buildContextPath := constants.BuildContextDir
-	if err := util.UnpackTarFromGCSBucket(bucket, buildContextPath); err != nil {
+	var contextExecutor buildcontext.BuildContext
+
+	// if no prefix use Google Cloud Storage as default for backwards compability
+	if !strings.Contains(srcContext, "://") {
+		if bucket != "" {
+			srcContext = fmt.Sprintf("gs://%s", bucket)
+		} else if bucket == "" {
+			srcContext = fmt.Sprintf("dir://%s", srcContext)
+		}
+	}
+
+	// if no context is set, add default file context.tar.gz
+	if !strings.HasSuffix(srcContext, ".tar.gz") {
```

Comment:
Since this should only happen when pulling from a GCS bucket, we should probably move this logic into `gcs.go`

Reply:
I’d leave this here. That way people can rely on if they don’t specify an object path we go with the default behavior: use context.tar.gz from the bucket root. If you do specify a full path: we use that.